okhttp/src/main/java/com/squareup/okhttp/CertificatePinner.java

```diff
@@ -100,6 +100,18 @@
  * complexity and limit your ability to migrate between certificate authorities.
  * Do not use certificate pinning without the blessing of your server's TLS
  * administrator!
+ * 
+ * <h3>Note about self-signed certificates</h3>
+ * {@link CertificatePinner} is second layer of certificate check,
```

Comment:
Can you say this with fewer words and with fewer `{@link }` tags? We try to keep our docs very short and to the point so that people read them. The first sentence is all that's really essential here.